### PR TITLE
Case refactor

### LINKED ISSUE #462: case templates bootstrapping
This follows up on #292 

The main problem is, that this now hard requires templates to be configured, but:
- no templates are available in a default install (only in testcontent)
- there's no (documented) way to create a template TTW

In short, if we follow Maarten's procedure, and install an empty Plone Intranet site, this results in a broken product. 
1. We need documentation on how to configure templates from scratch.
2. Currently you cannot create a template TTW without a template...
   
   (a.) either we make it possible to bootstrap TTW, e.g. with sane defaults
   
   (b.) or we hide all "Cases" functionality unless templates are available (from a custom setup handler)

My guess is that 2a. is easier than 2b. It's also more elegant since 2b. requires integrators to write custom code before they can even use this part of the product.

### PR DESCRIPTION
**Pull request by gyst**
This PR makes cases in Plone Intranet releasable for Venus. Fixes:
- provide a properly secured workflow
- provide a good demo case
- provide a good default case template in the default install (fixes #462)
- make it possible for non-admins to create a case from a template
- switch todos from 'planned' to 'open' even on protected milestone transitions (replaces #758)
- activate workflow guard on forward state transitions (replaces #517)
- sanitize metromap logic
- provide test coverage on the above

### CODE REVIEW

**Comment by gyst** [comment]
Remaining failure is a Heisenbug on unrelated functionality.
Jenkins please test this.

**Comment by gyst** [comment]
Jenkins please test this test please.

File: src/ploneintranet/workspace/adapters.py [hunk 1]
```diff
@@ -271,34 +271,64 @@ def metromap_sequence(self):
...
             sequence[state] = {
                 'title': _(cwf.states.get(state).title),
                 'transition_id': next_transition,
-                'transition_title': '',
-                'reopen_transition': reopen and mmap['reopen_transition'],
-                'enabled': next_transition in next_available_transition,
+                'transition_title': transition_title,
+                'reopen_transition': reopen_transition,
+                'is_current': is_current,
                 'finished': finished,
```
Comment (gyst) [inline src/ploneintranet/workspace/adapters.py:330]: This replaces `enabled` with `is_current`. @cillianderoiste What I've done in Plone Intranet, you may need to do in ikath:
`grep -r enabled . | cut -f1 -d: | grep -v Binary | uniq | xargs sed -i -e 's/enabled/is_current/g'`

**Comment by gyst** [comment]
@pilz @cillianderoiste I've done a lot of work fixing up the case implementation for Plone Intranet OOTB. This is now ready for review and merge.

**Comment by gyst** [comment]
@pilz Can I please have this merged? It's a Venus blocker, I put a lot of work in it, and it risks coming into conflict with any change @cillianderoiste does in cases.

**Comment by pilz** [comment]
We are still investigating impact.

> Am 29.09.2015 um 10:55 schrieb Guido Stevens notifications@github.com:
> 
> @pilz https://github.com/pilz Can I please have this merged? It's a Venus blocker, I put a lot of work in it, and it risks coming into conflict with any change @cillianderoiste https://github.com/cillianderoiste does in cases.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/ploneintranet/ploneintranet/pull/773#issuecomment-143993580.

## 

Alexander Pilz
SYSLAB.COM GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
http://www.syslab.com - Amtsgericht Muenchen, HRB 135057
Steuernummer 143/184/50154 - Ust.-ID: DE212842815

**Comment by gyst** [comment]
Main impact should be the variable name change from 'enabled' to 'is_current'. Other than that it's sanitizing logic (mostly the metromap calculation and the todo reopening), adding test coverage, and creating the templates for PI.

**Comment by gyst** [comment]
Build 965 failure seems to be another example of Jenkins PR leakage.
Jenkins please test this.

**Comment by gyst** [comment]
Jenkins please test this.

**Comment by gyst** [comment]
Jenkins please test this test please.

**Comment by cillianderoiste** [comment]
I've checked through this, and tested it locally. It looks great and I'm happy to merge. I guess we ca
(EDIT: I submitted by accident) ... I guess we can wait for jenkins to pass before merging.

**Comment by gyst** [comment]
This is too weird for words. Build 973 overview reports 5 errors, but the actual console output shows all tests are green.

**Pull request merged**